### ISSUE TITLE
"uv add PySide6" fails when the latest version doesn't have a wheel for the current platform

### ISSUE THREAD

**Issue by borco** [astral-sh/uv#10056]
At the time of writing this, the latest version of `PySide6` is `6.8.1.1`. I assume this is a patch release only for macos, as there is only one wheel for macos in it.

When I try to install `PySide6` on Windows, I get an error that no source or wheel is available:

```bash
$ uv add pyside6
Resolved 5 packages in 6ms
error: Distribution `pyside6==6.8.1.1 @ registry+https://pypi.org/simple` can't be installed 
because it doesn't have a source distribution or wheel for the current platform
```

However, the older release `6.8.1` contains wheels for many other platforms. If I blacklist the `6.8.1.1` version, then installing on Windows works:

```bash
$ uv add "pyside6>=6.8.1,!=6.8.1.1"
Resolved 5 packages in 257ms
Installed 4 packages in 1.05s
 + pyside6==6.8.1
 + pyside6-addons==6.8.1
 + pyside6-essentials==6.8.1
 + shiboken6==6.8.1
```

Is it possible to automatically detect such a case and install from the older version without having to explicitly blacklist the latest version that only contains wheels for other platform(s)?

[PySide 6.8.1.1 files](https://pypi.org/project/PySide6/6.8.1.1/#files)

[PySide 6.8.1 files](https://pypi.org/project/PySide6/6.8.1/#files)

![Image](https://github.com/user-attachments/assets/56ee276a-60d1-4cb2-ab9e-aa7ec35b19f0)

![Image](https://github.com/user-attachments/assets/a9f461af-7d63-413a-9902-c7fcafd7d165)

**Comment by charliermarsh** [comment]
This is tracked in https://github.com/astral-sh/uv/issues/9711 (though this is really a package issue -- they should be uploading a consistent set of wheels, the system isn't designed to be used this way).

**Comment by cbrnr** [comment]
I agree that this is really a packaging issue, and the list of packages that misuse the system is kind of shocking! Is there any workaround for this particular case that we could use in the meantime?

Also, maybe we should let the packagers know that they shouldn't really be doing it like that? Has anyone done that yet?

**Comment by cbrnr** [comment]
They've uploaded the missing packages so this issue can be closed.

**Comment by charliermarsh** [comment]
Oh great, thank you!
